src/amo/components/FeaturedAddonReview/index.js

```diff
@@ -0,0 +1,136 @@
+/* @flow */
+import * as React from 'react';
+import NestedStatus from 'react-nested-status';
+import { connect } from 'react-redux';
+import { withRouter } from 'react-router-dom';
+import { compose } from 'redux';
+
+import { fetchReview } from 'amo/actions/reviews';
+import AddonReviewCard from 'amo/components/AddonReviewCard';
+import { selectReview } from 'amo/reducers/reviews';
+import { withFixedErrorHandler } from 'core/errorHandler';
+import translate from 'core/i18n/translate';
+import log from 'core/logger';
+import Card from 'ui/components/Card';
+import type { UserReviewType } from 'amo/actions/reviews';
+import type { AppState } from 'amo/store';
+import type { ErrorHandlerType } from 'core/errorHandler';
+import type { AddonType } from 'core/types/addons';
+import type { I18nType } from 'core/types/i18n';
+import type { DispatchFunc } from 'core/types/redux';
+import type { ReactRouterLocationType } from 'core/types/router';
+
+import './styles.scss';
+
+type Props = {|
+  addon: AddonType | null,
+  reviewId: number,
+|};
+
+type InternalProps = {|
+  ...Props,
+  dispatch: DispatchFunc,
+  errorHandler: ErrorHandlerType,
+  featuredReview?: UserReviewType,
+  i18n: I18nType,
+  loadingReview: boolean,
+  location: ReactRouterLocationType,
+|};
+
+export class FeaturedAddonReviewBase extends React.Component<InternalProps> {
+  constructor(props: InternalProps) {
+    super(props);
+    this.loadDataIfNeeded();
+  }
+
+  componentWillReceiveProps(nextProps: InternalProps) {
+    this.loadDataIfNeeded(nextProps);
+  }
+
+  loadDataIfNeeded(nextProps?: InternalProps) {
+    const {
+      dispatch,
+      errorHandler,
+      featuredReview,
+      loadingReview,
+      reviewId,
+    } = {
+      ...this.props,
+      ...nextProps,
+    };
+
+    if (errorHandler.hasError()) {
+      log.warn('Not loading data because of an error');
+      return;
+    }
+
+    if (!featuredReview && !loadingReview) {
+      dispatch(fetchReview({ reviewId, errorHandlerId: errorHandler.id }));
+    }
+  }
+
+  render() {
+    const { addon, errorHandler, featuredReview, location, i18n } = this.props;
+
+    const featuredReviewHeader = featuredReview
+      ? i18n.sprintf(i18n.gettext('Review by %(userName)s'), {
+          userName: featuredReview.userName,
+        })
+      : null;
+
+    const featuredReviewCard =
+      errorHandler.hasError() &&
+      errorHandler.capturedError.responseStatusCode === 404 ? (
+        <NestedStatus code={404}>
```

Comment:
BTW, I don't know if your previous comment about this was outdated but this is working as intended. I see a 404 response status.

Reply:
Why do we use `NestedStatus` here and not the `NotFound` component?

Reply:
The first attempt at this patch was to use a new top level component for the featured review but that approach didn't work. That approach would have made it easier to show a generic 404 page. I don't know if it's worth the effort of prop drilling to make it work in this new approach.